pydis_site/apps/content/resources/guides/python-guides/subclassing_context.md

```diff
@@ -0,0 +1,137 @@
+        if self.delete_after:
+            await interaction.delete_original_message()
+        self.stop() #stops the view and deletes the confirmation message
+
+    @discord.ui.button(label='Cancel', style=discord.ButtonStyle.red)
+    async def cancel(self, button: discord.ui.Button, interaction: discord.Interaction):
+        self.value = False #returns False if the cancel button is pressed
+        await interaction.response.defer()
+        if self.delete_after:
+            await interaction.delete_original_message()
+        self.stop() #stops the view and deletes the confirmation
+
+# We have made two buttons, one for Confirm and one for Cancel.
+
+class CustomContext(commands.Context):
+    # create new Context methods here. For example:
+
+    async def send_embed(self, message, **kwargs):
+        return await self.send(embed=discord.Embed(description=message, colour=discord.Color.blurple(), **kwargs))
+        # This is a shortcut method to sending a custom embed. Pre-set values can be put here, for example the color kwarg in the example above means it has a pre-set color of blurple
+
+    async def send_error_embed(self, message, **kwargs):
+        return await self.send(embed=discord.Embed(description=message, colour=discord.Color.red(), **kwargs))
+        # embed with pre-set color of red
+
+
+    async def send_sucess_embed(self, message, **kwargs):
+        return await self.send(embed=discord.Embed(description=message, colour=discord.Color.green(), **kwargs))
+        # embed with pre-set color of green
+
+    # our custom prompt function using buttons  
+    async def prompt(self,message: str, *, timeout: float = 60.0, delete_after: bool = True, author_id: Optional[int] = None,) -> Optional[bool]:
+        """
+        Parameters
+        message: str
+            The message to show along with the prompt.
+
+        timeout: float (Defaults to 60 seconds)
+            How long to wait before stopping the view
+        delete_after: bool (Defaults to True)
+            Whether to delete the confirmation message after we're done.
+
+        author_id: Optional[int] (Defaults to ctx.author)
+            The member who should respond to the prompt.
+        -------
+        Returns  
+            True if the confirm button is pressed,
+            False if cancel button is pressed,
+            None if automatically denied due to no response
+        """
+
+        author_id = author_id or self.author.id
+        # use the View we have made above and set the arguments
+        view = PromptView(
+            timeout=timeout,
+            delete_after=delete_after,
+            ctx=self,
+            author_id=author_id,
+        )
+
+        view.message = await self.send(message, view=view)
+        await view.wait()
+        return view.value #Can return True, False or None
+
+
+class Bot(commands.Bot):
+    def __init__(self):
+        super().__init__(
+            #key word arguments
+        )
```

Comment:
We're not passing anything here, and I don't think we need to override `__init__` in the first place, do we?

Reply:
I put a comment of `key word arguments` there in the init with the intention of increasing the need to go over the subclassing_bot article to understand how to subclass bot. If I were to add the key word arguments, in my opinion people are more likely to just copy paste without fully understanding (in line 12, I wanted to link the subclassing_bot file). But if this isn't a good reasoning I can change it.